Source/Utils/Inc/Utils/Serialisation/Reader.h

```diff
@@ -42,6 +42,18 @@ class Reader
 
 	virtual bool ReadProperty( const PropertyName& Name, std::string& Value ) = 0;
 
+	template< class TEnum >
+	auto ReadProperty( const PropertyName& Name, TEnum& Value ) -> typename std::enable_if< std::is_enum< TEnum >::value, bool >::type
+	{
+		typedef typename std::underlying_type< TEnum >::type TEnumType;
+
+		TEnumType Out;
```

Comment:
Wonder if this could be named something a little more descriptive? `EnumValue` perhaps?

Reply:
Sure, `EnumValue` makes sense to me!